azurerm/resource_arm_action_group.go

```diff
@@ -0,0 +1,320 @@
+
+	resp, err := client.Get(ctx, resGroup, name)
+	if err != nil {
+		if response.WasNotFound(resp.Response.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error getting action group %s (resource group %s): %+v", name, resGroup, err)
+	}
+
+	d.Set("name", name)
+	d.Set("resource_group_name", resGroup)
+	if location := resp.Location; location != nil {
+		d.Set("location", azureRMNormalizeLocation(*location))
+	}
+
+	d.Set("short_name", *resp.GroupShortName)
+	d.Set("enabled", *resp.Enabled)
+
+	if err = d.Set("email_receiver", flattenActionGroupEmailReceiver(resp.EmailReceivers)); err != nil {
+		return fmt.Errorf("Error setting `email_receiver` of action group %s (resource group %s): %+v", name, resGroup, err)
+	}
+
+	if err = d.Set("sms_receiver", flattenActionGroupSmsReceiver(resp.SmsReceivers)); err != nil {
+		return fmt.Errorf("Error setting `sms_receiver` of action group %s (resource group %s): %+v", name, resGroup, err)
+	}
+
+	if err = d.Set("webhook_receiver", flattenActionGroupWebHookReceiver(resp.WebhookReceivers)); err != nil {
+		return fmt.Errorf("Error setting `webhook_receiver` of action group %s (resource group %s): %+v", name, resGroup, err)
+	}
+
+	flattenAndSetTags(d, resp.Tags)
+
+	return nil
+}
+
+func resourceArmActionGroupDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).actionGroupsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return fmt.Errorf("Error parsing action group resource ID \"%s\" during delete: %+v", d.Id(), err)
+	}
+	resGroup := id.ResourceGroup
+	name := id.Path["actionGroups"]
+
+	resp, err := client.Delete(ctx, resGroup, name)
+	if err != nil {
+		if response.WasNotFound(resp.Response) {
+			return nil
+		}
+		return fmt.Errorf("Error deleting action group %s (resource group %s): %+v", name, resGroup, err)
+	}
+
+	return nil
+}
+
+func expandActionGroupEmailReceiver(d *schema.ResourceData) *[]insights.EmailReceiver {
+	v, ok := d.GetOk("email_receiver")
+	if !ok {
+		return nil
+	}
+
+	receivers := make([]insights.EmailReceiver, 0)
+	for _, receiverValue := range v.([]interface{}) {
+		val := receiverValue.(map[string]interface{})
+
+		receiver := insights.EmailReceiver{
+			Name:         utils.String(val["name"].(string)),
+			EmailAddress: utils.String(val["email_address"].(string)),
+		}
+
+		receivers = append(receivers, receiver)
+	}
+	return &receivers
+}
+
+func flattenActionGroupEmailReceiver(receivers *[]insights.EmailReceiver) []interface{} {
+	result := make([]interface{}, 0)
+	if receivers != nil {
+		for _, receiver := range *receivers {
+			val := make(map[string]interface{}, 0)
+
+			val["name"] = *receiver.Name
+			val["email_address"] = *receiver.EmailAddress
+
+			result = append(result, val)
+		}
+	}
+	return result
+}
+
+func expandActionGroupSmsReceiver(d *schema.ResourceData) *[]insights.SmsReceiver {
+	v, ok := d.GetOk("sms_receiver")
+	if !ok {
+		return nil
```

Comment:
Unless there is a reason to return nil here, can we return an empty array as above?

Reply:
Done.